Razor/Scripts/Engine/Interpreter.cs

```diff
@@ -1075,9 +1075,8 @@ private IComparable EvaluateBinaryOperand(ref ASTNode node)
                         var arg = Interpreter.GetVariable(node.Lexeme);
                         if (arg != null)
                         {
-                            // We don't want to use arg.AsString() here because it will attempt
-                            // to evaluate aliases and may get us into an infinite loop.
-                            val = node.Lexeme;
+
+                            val = op == ASTNodeType.AS ? node.Lexeme : arg.AsString();
```

Comment:
For operands other then AS we should get value, but for AS we need to get only label of variable.